### ISSUE TITLE
[rl] PpgAuxValueHead has no num_actions(), agent infers it by len/total_steps division

### ISSUE THREAD

**Issue by anthonytorlucci** [anthonytorlucci/rlevo#512]
- **Files:** `crates/rlevo-reinforcement-learning/src/algorithms/ppg/ppg_policy.rs`, `crates/rlevo-reinforcement-learning/src/algorithms/ppg/ppg_agent.rs`, `crates/rlevo-reinforcement-learning/src/algorithms/ppg/policies/categorical.rs`
- **Body:** Confirmed current (2026-07-20). `ppg_agent.rs:704`:

```rust
let num_actions = old_logits_flat.len() / total_steps;
```

The agent is generic over `P: PpoPolicy<B, DB> + PpgAuxValueHead<B, DB>` and recovers `num_actions` by integer division of a flat logits buffer, because `PpgAuxValueHead` exposes no width accessor. Note the concrete `PpgCategoricalPolicyHead` already has a `pub fn num_actions(&self) -> usize` inherent method (`categorical.rs:93-95`) — it's just not reachable through the trait bound the agent is written against, so the agent can't use it and falls back to the division inference instead.

Fix: add `fn num_actions(&self) -> usize;` to `PpgAuxValueHead` (mirrors `PpoPolicy::action_dim`), have `PpgCategoricalPolicyHead`'s existing inherent method satisfy it, and replace the `len / total_steps` line in `ppg_agent.rs:704` with `self.policy().num_actions()`.
- **Severity:** low-medium (guarded one line above by a `total_steps == 0` check per review, so not a live division-by-zero; this is a fragile-inference/API-completeness gap, not a crash)
- **Source:** Code review 2026-06-30
